File: scripts/algorithms/Zigzag Conversion/Zigzag Conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if (numRows==1)return s;
        StringBuilder builder = new StringBuilder();
        for (int i=1;i<=numRows;i++){
            int ind = i-1;
            boolean up = true;
            while (ind < s.length()){
                builder.append(s.charAt(ind));
                if (i==1){
                    ind += 2*(numRows-i);
                } else if (i==numRows){
                    ind += 2*(i-1);
                } else {
                    ind += up ? 2*(numRows-i) : 2*(i-1);
                    up=!up;
                }
            }
        }
        return builder.toString();
    }
}

File: algorithms/Z/Zigzag Conversion/Zigzag Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        vector<vector<char>> v(numRows);
        int j=0,t=1;
        if(numRows ==1)
            return s;
        for(int i=0;i<s.size();i++)
        {
            v[j].push_back((char)s[i]);
            if(j==numRows-1 )
                t=0;
            else if(j==0)
                t=1;
            if(t)
                j++;
            else
                j--;
            
        }
        string x="";
        for(int i=0;i<numRows;i++)
        {
            for(int j=0;j<v[i].size();j++)
                x.push_back(v[i][j]);
        }
        return x;
    }
};
